packages/xo-web/src/xo-app/hub/resource.js

```diff
@@ -0,0 +1,258 @@
+              <ResourceForm
+                install
+                multi
+                poolPredicate={installPoolPredicate}
+                {...props}
+              />
+            ),
+            header: (
+              <span>
+                <Icon icon='add-vm' /> {name}
+              </span>
+            ),
+            size: 'medium',
+          })
+
+          setHubInstallLoadingState({
+            ...hubInstallLoadingState,
+            [id]: true,
+          })
+          for (const pool of resourceParams.pools) {
+            try {
+              await downloadAndInstallResource({
+                namespace,
+                id,
+                version,
+                sr: pool.default_SR,
+              })
+              success('XVA import', _('hubSuccessfulInstallMsg'))
+            } catch (_error) {
+              error('Error', _error.message)
+            }
+          }
+          setHubInstallLoadingState({
+            ...hubInstallLoadingState,
+            [id]: false,
+          })
+        }
+      },
+      async create(__, { name }) {
+        const {
+          isFromSources,
+          createPoolPredicate,
+          installedTemplates,
+        } = this.state
+        if (isFromSources) {
+          subscribeAlert()
+        } else {
+          const resourceParams = await form({
+            render: props => (
+              <ResourceForm poolPredicate={createPoolPredicate} {...props} />
+            ),
+            header: (
+              <span>
+                <Icon icon='add-vm' /> {name}
+              </span>
+            ),
+            size: 'medium',
+          })
+          const { $pool } = resourceParams.pool
+          const { id } = find(installedTemplates, ['$pool', $pool])
+          this.props.router.push(`/vms/new?pool=${$pool}&template=${id}`)
+        }
+      },
+      async deleteTemplates(__, { name }) {
+        const { createPoolPredicate } = this.state
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm
+              delete
+              multi
+              poolPredicate={createPoolPredicate}
+              {...props}
+            />
+          ),
+          header: (
+            <span>
+              <Icon icon='vm-delete' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+        const _templates = filter(this.state.installedTemplates, template =>
+          find(resourceParams.pools, ['$pool', template.$pool])
+        )
+        await deleteTemplates(_templates)
+      },
+      updateSelectedInstallPools(_, selectedInstallPools) {
+        return {
+          selectedInstallPools,
+        }
+      },
+      updateSelectedCreatePool(_, selectedCreatePool) {
+        return {
+          selectedCreatePool,
+        }
+      },
+      redirectToTaskPage() {
+        this.props.router.push('/tasks')
+      },
+    },
+    computed: {
+      isFromSources: () => +process.env.XOA_PLAN > 4,
```

Comment:
You can use [`getXoaPlan`](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/utils.js#L152-L167) instead.